operator/pkg/apis/core/customize/v1alpha1/controllerreconciler_types.go

```diff
@@ -41,6 +41,9 @@ type NamespacedControllerReconcilerSpec struct {
 	// If not specified, the default will be Token Bucket with qps 20, burst 30.
 	// +optional
 	RateLimit *RateLimit `json:"rateLimit,omitempty"`
+	// PprofConfigures the debug endpoint on the service.
+	// +optional
+	PprofConfig *PprofConfig `json:"pprofConfig,omitempty"`
```

Comment:
you can probably argue YAGNI against this but how about we sub class the Pprof config under something like `debugConfig` and we can add more debugging controls like otel, etc.

Reply:
My personal take is if we think we will have many of these (say more than 5) then this organization would make a lot of sense.
Right now we have 1 and we'll see if we feel the need for the second.
It its just 2 it won't hurt but I'm not convinced its worth the effort of changing.
I may come to regret that.

Reply:
Since this is an alpha API, we can always make improvements in the future if needed.